lib/Parser/DocumentParser.php

```diff
@@ -479,11 +490,17 @@ private function flush(): void
                     /** @var string[] $lines */
                     $lines = $this->buffer->getLines();
 
-                    $blockNode = $this->nodeFactory->createBlockNode($lines);
+                    $node = $this->nodeFactory->createBlockNode($lines);
 
-                    $document = $this->parser->getSubParser()->parseLocal($blockNode->getValue());
+                    // This means we are in an indented area that is not a code block.
+                    // If we're NOT in a directive, then this must be a blockquote.
+                    // If we ARE in a directive, allow the directice to convert
+                    // the BlockNode into what it needs
+                    if ($this->directive === null) {
+                        $document = $this->parser->getSubParser()->parseLocal($node->getValue());
 
-                    $node = $this->nodeFactory->createQuoteNode($document);
+                        $node = $this->nodeFactory->createQuoteNode($document);
+                    }
```

Comment:
This represents the key bug fix that allows directives to start returning false from `wantsCode()`. 

This is a edge case BC break (bit it's a fix): if a directive's `wantsCode()` currently returns `false`, then previously we would pass that directive a `QuoteNode` where the "value" is this `$document` variable. Now you would be passed a `BlockNode` where the value is just the string inside the code block.

But, the idea is that IF you want to actually "parse" the "content/value" of your directive as rst, then you should use a "sub parser". And, as far as I can tell, that's precisely the purpose of the `SubDirective` class: it takes in the raw string content/value, then parses it further. THAT is the proper location for this type of logic, I believe.